### PR TITLE
Update build file and the Loaders

### PR DESCRIPTION
**Pull request by roideuniverse**

### CODE REVIEW

File: app/src/main/java/com/carrotcreative/cream_example/app/cache/params/GithubUserLoaderParams.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    public String getUserId()
+    {
+        return mUserId;
+    }
+
+    @Override
+    public String getIdentifier()
+    {
+        return mUserId;
```
Comment (BrandonRomano) [inline app/src/main/java/com/carrotcreative/cream_example/app/cache/params/GithubUserLoaderParams.java]: Even though this is a trivial example, this should probably use `getUserId()` instead, as that's how we use it most of the time in production.

**Comment by BrandonRomano** [comment]
Where did we update the CREAM lib here?

File: app/build.gradle [hunk 2]
```diff
@@ -12,14 +12,14 @@ android {
...
-            runProguard false
+            minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.txt'
         }
     }
 }
 
 dependencies {
-    compile fileTree(dir: 'libs', include: ['*.jar'])
+    compile fileTree(include: ['*.jar'], dir: 'libs')
```
Comment (BrandonRomano) [inline app/build.gradle:22]: Why did we flip these, out of curiosity?

File: app/src/main/java/com/carrotcreative/cream_example/app/cache/loaders/GithubRepoLoader.java [hunk 3]
```diff
@@ -66,13 +67,13 @@ public void failure(RetrofitError error) {
      * See "Multiple params in API Call"
      * section in GithubRepoLoader
      */
-    public class RepoDefinition
+    public class RepoDefinition implements LoaderParams
```
Comment (BrandonRomano) [inline app/src/main/java/com/carrotcreative/cream_example/app/cache/loaders/GithubRepoLoader.java]: Can we change this to RepoParams here too? (To match with Cream's README)

File: app/build.gradle (hunk 2 above)
Comment (roideuniverse) [inline app/build.gradle:22]: I didn't do it. Android Studio did that. :smile_cat: 
@BrandonRomano : I guess its Android Studio's default. Do you want me to switch it back ?

File: build.gradle [hunk 4]
```diff
@@ -5,7 +5,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:0.9.+'
+        classpath 'com.android.tools.build:gradle:1.2.3'
```
Comment (BrandonRomano) [inline build.gradle:8]: Hmm.. I see that we've added the submodule -- but I still don't see it being included anywhere!

Comment (roideuniverse) [inline build.gradle:8]: Maybe i didn't do it right - but its here a9da3e62c9b9818e3a5a02adc238b67449bab033

File: app/build.gradle (hunk 2 above)
Comment (BrandonRomano) [inline app/build.gradle:22]: Nope, it's not a problem -- just was curious if this was an intentional decision!

File: build.gradle (hunk 4 above)
Comment (BrandonRomano) [inline build.gradle:8]: Ah, okay... So you've added it as a git submodule -- but in addition to that you actually have to include the project in Gradle as a module!

File: app/build.gradle (hunk 2 above)
Comment (roideuniverse) [inline app/build.gradle:22]: It was - I didn't revert is back only because it looked like Android Studio preferred that.

File: build.gradle (hunk 4 above)
Comment (roideuniverse) [inline build.gradle:8]: Yup, but I they were already added from beforehand.
https://github.com/carrot/cream-example/blob/master/settings.gradle
https://github.com/carrot/cream-example/blob/master/app/build.gradle

File: app/build.gradle (hunk 2 above)
Comment (BrandonRomano) [inline app/build.gradle:22]: Okay cool :+1:

**Comment by BrandonRomano** [comment]
:+1:  from my end -- @roideuniverse check out the code I just wrote and make sure you also are in agreement with everything.  If you're cool with it feel free to merge

**Pull request merged**